cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs

```diff
@@ -12,48 +12,74 @@ import           Cardano.Prelude
 import           Cardano.Config.Shelley.ColdKeys (KeyError, KeyRole (..), OperatorKeyRole (..),
                     readVerKey)
 
+import           Cardano.Chain.Common (Lovelace, unsafeGetLovelace)
+import           Cardano.CLI.Ops (CliError (..))
+import           Cardano.CLI.Shelley.Run.KeyGen (runGenesisKeyGenDelegate, runGenesisKeyGenGenesis)
+import           Cardano.CLI.Shelley.Parsers (GenesisDir (..), OpCertCounterFile (..),
+                    SigningKeyFile (..), VerificationKeyFile (..))
+import           Cardano.Config.Shelley.Genesis
+
 import           Control.Monad.Trans.Except (ExceptT)
 import           Control.Monad.Trans.Except.Extra (firstExceptT, handleIOExceptT, hoistEither)
 
 import qualified Data.Aeson as Aeson
 import           Data.Aeson.Encode.Pretty (encodePretty)
 import qualified Data.ByteString.Lazy.Char8 as LBS
 import           Data.Char (isDigit)
-import           Data.Coerce (coerce)
 import           Data.Map.Strict (Map)
 import qualified Data.Map.Strict as Map
 import qualified Data.Text as Text
 import           Data.Time.Clock (NominalDiffTime, UTCTime, addUTCTime, getCurrentTime)
 
-import           Cardano.Chain.Common (Lovelace, unsafeGetLovelace)
-import           Cardano.CLI.Ops (CliError (..))
-import           Cardano.CLI.Shelley.Parsers (GenesisDir (..))
-import           Cardano.Config.Shelley.Genesis
-
 import           Ouroboros.Consensus.BlockchainTime (SystemStart (..))
 import           Ouroboros.Consensus.Shelley.Protocol (TPraosStandardCrypto)
 
 import qualified Prelude
 
-import           Shelley.Spec.Ledger.Keys (DiscKeyHash (..), GenKeyHash, KeyHash)
+import           Shelley.Spec.Ledger.Keys (GenKeyHash, KeyHash)
 import qualified Shelley.Spec.Ledger.Keys as Ledger
 
-import           System.Directory (listDirectory)
+import           System.Directory (createDirectoryIfMissing, listDirectory)
 import           System.FilePath ((</>), takeExtension)
 
-runGenesisCreate :: GenesisDir -> Maybe SystemStart -> Lovelace -> ExceptT CliError IO ()
-runGenesisCreate (GenesisDir gendir) mStart amount = do
+runGenesisCreate :: GenesisDir -> Word -> Maybe SystemStart -> Lovelace -> ExceptT CliError IO ()
+runGenesisCreate (GenesisDir gendir) count mStart amount = do
   start <- maybe (SystemStart <$> getCurrentTimePlus30) pure mStart
   template <- readShelleyGenesis (gendir </> "genesis.spec.json")
+
+  forM_ [ 1 .. count ] $
+    createRequiredKeys gendir
```

Comment:
Ah, so now this means the behaviour is it always automatically makes the keys (and then finds them again below).

Perhaps we can easily allow for both manually creating the keys and then automatically making the genesis, and also this do-it-all-in-one-go behaviour of making the keys and the genesis.

If we were to make the default for the `--genesis-delegates` be 0, then the `forM_ [1 .. 0]` will do nothing by default, and will create N keys when specified, and so we can get both semi and fully automatic for cheap.